server/routes/shared_api/node.py

```diff
@@ -45,3 +51,59 @@ def get_property_value(direction):
     prop = request.json['prop']
   response = fetch.raw_property_values(dcids, prop, direction == 'out')
   return Response(json.dumps(response), 200, mimetype='application/json')
+
+
+@bp.route('/propvals', methods=['GET', 'POST'])
+def expression_property_value():
+  """
+  Returns the property values for given node dcids and a relation expression for
+  a property (https://docs.datacommons.org/api/rest/v2#relation-expressions)
+  """
+  dcids = request.args.getlist('dcids')
+  if not dcids:
+    dcids = request.json['dcids']
+  prop_expression = request.args.get('propExpr')
+  if not prop_expression:
+    prop_expression = request.json['propExpr']
+  if not dcids:
+    return 'error: must provide a `dcids` field', 400
+  if not prop_expression:
+    return 'error: must provide a `propExpr` field', 400
+  response = get_expression_property_value(dcids, prop_expression)
+  return Response(json.dumps(response), 200, mimetype='application/json')
+
+
+def get_expression_property_value(dcids, expression):
+  """
+  Returns the property values for given node dcids and a relation expression for
+  a property (https://docs.datacommons.org/api/rest/v2#relation-expressions)
+  """
+  # Get a list of single property relation expressions (the expression passed to
+  # this function can chain multiple relation expressions)
+  expression_matches = re.finditer(_PROPERTY_EXPRESSION_RE, expression)
```

Comment:
Can you give an example of the chaining here? Given the complexity of the logic, I feel this function worth to have a test.

Reply:
added a test that has a chaining example